tests/lib/scan/scan_test.py

```diff
@@ -0,0 +1,168 @@
+        self.addCleanup(self.patcher1.stop)
+
+        self.patcher2 = mock.patch(
+            "prowler.lib.scan.scan.update_audit_metadata", autospec=True
+        )
+        self.mock_update_audit_metadata = self.patcher2.start()
+        self.addCleanup(self.patcher2.stop)
+
+        self.patcher3 = mock.patch("prowler.lib.logger.logger", autospec=True)
+        self.mock_logger = self.patcher3.start()
+        self.addCleanup(self.patcher3.stop)
+
+        self.patcher4 = mock.patch(
+            "prowler.providers.common.provider.Provider.get_global_provider",
+            return_value=self.mock_provider,
+        )
+        self.mock_global_provider = self.patcher4.start()
+        self.addCleanup(self.patcher4.stop)
+
+        self.findings = [
+            generate_finding_output(
+                status="PASS",
+                status_extended="status-extended",
+                resource_uid="resource-123",
+                resource_name="Example Resource",
+                resource_details="Detailed information about the resource",
+                resource_tags="tag1,tag2",
+                partition="aws",
+                description="Description of the finding",
+                risk="High",
+                related_url="http://example.com",
+                remediation_recommendation_text="Recommendation text",
+                remediation_recommendation_url="http://example.com/remediation",
+                remediation_code_nativeiac="native-iac-code",
+                remediation_code_terraform="terraform-code",
+                remediation_code_other="other-code",
+                remediation_code_cli="cli-code",
+                compliance={"compliance_key": "compliance_value"},
+                categories="category1,category2",
+                depends_on="dependency",
+                related_to="related finding",
+                notes="Notes about the finding",
+            )
+        ]
+
+        self.mock_execute.side_effect = [self.findings, self.findings]
+
+    def test_scan(self):
+        checks_to_execute = {"accessanalyzer_enabled", "ec2_instance_public_ip"}
+        custom_checks_metadata = {}
+
+        # Create a Scan object
+        scan = Scan(self.mock_provider, checks_to_execute)
+
+        # Execute the scan
+        results = list(scan.scan(custom_checks_metadata))
+
+        # Verify that execute was called twice
+        assert self.mock_execute.call_count == 2
+
+        assert len(results) == 2
+        assert results[0][1] == self.findings
+        assert results[1][1] == self.findings
+
+        # verify that update_audit_metadata was called twice
+        assert self.mock_update_audit_metadata.call_count == 2
+
+        # verify that the progress is 1.0
+        assert scan.progress == 1.0
+        assert scan._number_of_checks_completed == 2
+
+        # Verify that the findings are correct
+        assert scan.findings == self.findings + self.findings
```

Comment:
Why this?

Reply:
I've mocked the same finding twice, I could create a list that holds the two findings (wich are the same) or make it this way